src/components/TransactionFlow/displays/TokenTransferTable.tsx

```diff
@@ -0,0 +1,169 @@
+    }
+  }
+`;
+
+const Row = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  min-height: 60px;
+  width: 100%;
+  gap: ${SPACING.XS};
+  & > *:last-child {
+    flex: 1;
+  }
+  @media screen and (max-width: ${BREAK_POINTS.SCREEN_SM}) {
+    flex-direction: column;
+    align-items: flex-start;
+    position: relative;
+    width: 100%;
+    border-bottom: 1px solid ${COLORS.GREY_ATHENS};
+    padding: ${SPACING.SM} 0 ${SPACING.SM} 0;
+    & > *:last-child {
+      align-self: flex-end;
+      flex: 1;
+      border-bottom: 0px;
+    }
+  }
+`;
+
+const SText = styled(Text)`
+  margin: ${SPACING.NONE} ${SPACING.XS} ${SPACING.NONE} ${SPACING.NONE};
+`
+
+const STransferAmounts = styled(Text)`
+  /* Night Mode/ Accent */
+  padding: 1px 4px;
+  background: #8F8F8F;
+  font-weight: bold;
+  border-radius: 5px;
+  color: white;
+  margin: ${SPACING.NONE};
+`
+
+const LabelBox = styled(Box)`
+  gap: ${SPACING.XS};
+`
+
+const SMoreIconWrapper = styled.div`
+  flex: 1;
+  display: flex;
+  justify-content: flex-end;
+`
+
+const TokenTransferTable = ({ valueTransfers, settings }: Props) => {
+  const [isExpanded, setExpanded] = useState(false)
+  const tokenTransfers = valueTransfers.filter(t => t.asset.type === 'erc20')
+  return (
+    <ERC20Box>
+      <LabelBox variant="rowAlign">
+        <Text variant="subHeading" mb={SPACING.NONE}>{'Tokens Transferred: '}</Text>
+        <STransferAmounts>{`${tokenTransfers.length}`}</STransferAmounts>
+        <SMoreIconWrapper>
+          <Icon
+            type="more"
+            alt="More"
+            isExpanded={isExpanded}
+            rotate90Deg={true}
+            onClick={() => {
+              setExpanded(!isExpanded);
+            }}
+            height="1em"
+            color={COLORS.BLUE_BRIGHT}
+          />
+        </SMoreIconWrapper>
+      </LabelBox>
+      {isExpanded && <Table>
+        <Body>
+          {tokenTransfers.map((transfer, idx) => (
+            <Row key={idx}>
+              <Box variant='rowAlign'>
+                <SText>
+                  {`${translateRaw('RECENT_TRANSACTIONS_FROM_ADDRESS')}:`}
+                </SText>
+                <EthAddress
+                  address={toChecksumAddress(transfer.from)}
+                  truncate={true}
+                  textColor={COLORS.BLUE_SKY}
+                  iconColor={COLORS.BLUE_SKY}
+                  fontSize="16px"
+                />
+              </Box>
+              <Box variant='rowAlign'>
+                <SText>
+                  {`${translateRaw('RECENT_TRANSACTIONS_TO_ADDRESS')}:`}
+                </SText>
+                <EthAddress
+                  address={toChecksumAddress(transfer.to)}
+                  truncate={true}
+                  textColor={COLORS.BLUE_SKY}
+                  iconColor={COLORS.BLUE_SKY}
+                  fontSize="16px"
+                />
+              </Box>
+              {transfer.amount != '' ? <Amount
```

Comment:
When is `amount` an empty string? Can we do something safer than that?

Reply:
`amount` is an empty string only when we're using "Unknown Asset" "Unknown ERC20" or "Unknown ERC721" to describe the value transfer. we can instead add a prop that is `isUnknown` or something, but then we have 2 things that are different between unknown and known value transfers.

Reply:
Could we have amounts be undefined instead of empty strings in this case then?